samples/sensor/sht3xd/dts_fixup.h

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2018 Peter Bigot Consulting, LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_HAS_DTS_I2C)
+
+#define DT_SHT3XD_NAME ""
+#define DT_SHT3XD_I2C_ADDR 0
+#define DT_SHT3XD_I2C_MASTER_DEV_NAME ""
```

Comment:
I'd like to avoid this pattern for samples as it will cause users to think this sample will work on their board automatically, when in reality they need to create a dts overlay to define how the sensor is connected to the mcu. With the fixup, the compile will succeed but the application will fail at runtime. We should instead limit CI to build this sample only for board(s) that support it.

You also need to add a sample.yaml file to tell CI which boards and configurations to build the sample for. See `samples/sensor/fxos8700/sample.yaml` for an example.

Reply:
Good points.  I've addressed them as best I can, with explanations below.

There are no boards that have this sensor natively, which is probably why there wasn't a sample before.  I use a breakout board.  I've put together something that does a build-only test for it, with support files that also serve as examples.  It may be possible to use a fixture but I couldn't figure out how to do it quickly.

I completely agree on sample dts_fixup bogosity, saving that that's how it's commonly being done now.  I've replaced it with the pattern that I've used in subsequent patches (not yet submitted as they depend on an unreviewed I2C PR).  I believe that approach provides a new user with a better clue as to what needs to be done.  If this pattern seems better then it would be nice to require it for all future samples, as consistently demonstrating best practices can reduce support requests....

Let me know your thoughts.  If you don't like either of those innovations, more detailed direction on what you'd like instead would be appreciated.

Reply:
You can add a dts overlay in the sample for the nrf51_ble400 board, similar to how `samples/sensor/ms5837/` defines an overlay for the nrf52840_pca10056 board.

If there is a an arduino shield board available with this sensor, a more generic solution would be to define a shield overlay like `boards/shields/x_nucleo_iks01a1/x_nucleo_iks01a1.overlay`.

Reply:
Yes, that's exactly what I did to make it compile: [nrf51_ble400.overlay](https://github.com/zephyrproject-rtos/zephyr/pull/11545/files#diff-b646cfa4611ed59bbd612054e5a7ed2b).  What I'd like to do with #11737 is add an alias to that, then get rid of the dts_fixup entirely.

Reply:
Why not use aliases in existing form? I used them for I2C eeprom slave #9943 as adding something that isn't bound to any board is hard with fixups. It would be great to create new drivers that use aliases not dts_fixups

Reply:
This isn't a new driver: it was added in 2016.  I'm new here and trying to figure out how much pushback people have when existing names that are visible to developers or users change.

What you suggest is exactly what I would love to do, as suggested in #11737.  Approval there seems to be blocked by big plans for an alternative solution, and disagreement on whether it's a proper use of aliases in device tree files.

What's here works and is consistent with current state-of-the-practice in Zephyr.  I'm happy to rework this if there's somebody who can make a decision on what's acceptable (I note nobody's yet been assigned this PR, though @mhelm has provided valuable feedback).  Absent that, my preference is to follow existing practice until alternatives have community consensus.